lib/mongodb.js

```diff
@@ -401,6 +403,12 @@ MongoDB.prototype.fromDatabase = function(modelName, data) {
       if (data[p] instanceof mongodb.Binary) {
         // Convert the Binary into String
         data[p] = data[p].toString();
+      // SPIKE: properties that should be converted to string from ObjectID
+      // SPIKE: do the same for toDatabase
+      } else if (p === 'id' || prop.mongodb && prop.mongodb.dataType.match(/objectid/i)) {
+        if (data[p]) {
+          data[p] = data[p].toString();
+        }
```

Comment:
Please try find a way how to use the existing code for converting `string` to `Decimal128`.